### PR TITLE
#66 work in progress make print_panic_info() public

### PR DESCRIPTION
**Pull request by kellpossible**
Draft implementation of #66

### CODE REVIEW

File: examples/panic_compose.rs [hunk 1]
```diff
@@ -0,0 +1,50 @@
+use color_eyre::eyre::Report;
```
Comment (kellpossible) [inline examples/panic_compose.rs:1]: Might be good to have a description for the purpose of this example? I'm guessing it's piping panic messages through to `tracing`?

File: src/writers.rs [hunk 2]
```diff
@@ -197,26 +198,24 @@ impl fmt::Display for EnvSection<'_> {
...
-                write!(
-                    f,
-                    "Ensure that you've setup an error layer and the versions match"
-                )?;
-            }
+        if let Some(&SpanTraceStatus::UNSUPPORTED) = self.0 {
+            writeln!(f, "Warning: SpanTrace capture is Unsupported.")?;
+            write!(
+                f,
+                "Ensure that you've setup an error layer and the versions match"
```
Comment (kellpossible) [inline src/writers.rs:217]: I know it's unrelated to this PR, but I found this message confusing when I first encountered it. Perhaps some guidance as to how to "setup an error layer", and what exactly an "error layer" is would be helpful to users who are unfamiliar with the ecosystem.

**Comment by kellpossible** [comment]
Nice one @yaahc looks like you did all the hard work!

File: examples/panic_compose.rs (hunk 1 above)
Comment (yaahc) [reply examples/panic_compose.rs:1]: yes, exactly

**Comment by yaahc** [comment]
> Nice one @yaahc looks like you did all the hard work!

heh, ya, I got nerd snipped a little and then immediately after a friend asked about passing panics to tracing so I decided to not wait around and test out some things myself :sweat_smile:

**Comment by kellpossible** [comment]
@yaahc anything I can do to help with the PR to get it through?

**Comment by yaahc** [comment]
ooh, I can take care of merging it, just slipped thru the cracks is all, ty for the reminder.

**Comment by kellpossible** [comment]
I've been doing some more testing with this, and some things which would be nice, to be able to control backtrace, spantrace, and color formatting for the eyre report display invocation that gets used for sending data to sentry.

image

At the moment I have (along with a bunch of other code setting up tracing): 

```rust
let (eyre_panic_hook, eyre_hook) = color_eyre::config::HookBuilder::default()
    .into_hooks();

eyre_hook.install()?;

std::panic::set_hook(Box::new(move |panic_info| {
    eprintln!("{}", eyre_panic_hook.panic_report(panic_info));
    sentry::integrations::panic::panic_handler(panic_info);
}));
```

**Comment by kellpossible** [comment]
Perhaps some of this can be achieved with the `sentry-eyre` integration I was playing with, I'll investigate a bit more today.

**Comment by kellpossible** [comment]
Okay I understand a bit more now, having backtrace capture enabled has performance implications, thus the decision to make it only enabled at startup via the environment variable makes a lot of sense. I guess the request for somehow being able to choose to format the error without the terminal colours still stands though.

**Comment by yaahc** [comment]
The performance implications are not really the same for `eyre::Report` and panic reports, so if we don't already have the ability to set independent rules for when to capture backtraces in panics and in errors I'd be happy to add that feature.

**Comment by kellpossible** [comment]
Okay, thanks! I'll have a bit more of a think about it and submit a formal feature request issue if it makes sense.

**Comment by yaahc** [comment]
I'm sorry this has taken so long, I've not had enough mental bandwidth to do much open source work recently, but I haven't forgotten about this issue, and I'll get this finalized as soon as I can.

**Comment by yaahc** [comment]
alright @kellpossible how does this look? I ended up re-implementing it mostly from scratch because there's another PR that I've been sitting on for a while as well which was going to create the mother of all merge conflicts with this PR thanks to my overzealous code reorganization last time. Dreading dealing with that is a large part of why this has taken so long, once again, sorry about that. Let me know if there are any issues with the current implementation, if not I'll cut a release with this right away.

**Comment by kellpossible** [comment]
No worries, figured something like that might be the case :sweat_smile: Sorry for my slow reply! I'll take a look at it today.

**Comment by kellpossible** [comment]
It seems to be working fine with the latest changes

**Pull request merged**